jobs/build-charms/charms.py

```diff
@@ -337,30 +346,40 @@ def promote(self, charm_entity, from_channel, to_channels):
         self._echo(
             f"Promoting :: {charm_entity:^35} :: from:{from_channel} to: {to_channels}"
         )
-        if from_channel == "unpublished":
+        if "unpublished" == from_channel:
             charm_status = self._unpublished_revisions(charm_entity)
         else:
             charm_status = [
                 row
                 for row in self.status(charm_entity)
-                if f"{row['Track']}/{row['Channel']}" == from_channel
+                if all(
+                    (
+                        row["Revision"],
+                        f"{row['Track']}/{row['Channel']}" == from_channel,
+                    )
+                )
```

Comment:
How about this?  Is this easier to read?

```suggestion
                if row["Revision"] and f"{row['Track']}/{row['Channel']}" == from_channel
```